### PR TITLE
Update dependencies and pins.

### PR DESCRIPTION
**Pull request by nc6**
- The `cardano-base`, `cardano-ledger`, `plutus` and `ouroboros-network`
  dependencies have been updated to recent versions.
- Since the downstream libraries have been updated to more recent
  index-state and hackage revisions, we synchronise this here.
- As a consequence of this update, we switch to a newer version of
  `aeson`, necessitating updates in `ekg-json` and `hedgehog-extras`.
- We switch to ghc8107, to be in sync with other packages.

### CODE REVIEW

File: cardano-api/src/Cardano/Api/TxMetadata.hs [hunk 1]
```diff
@@ -76,6 +76,8 @@ import           Cardano.Api.Eras
 import           Cardano.Api.Error
 import           Cardano.Api.HasTypeProxy
 import           Cardano.Api.SerialiseCBOR
+import qualified Data.Aeson.KeyMap as KeyMap
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/TxMetadata.hs:79]: These imports should be with `Data.Aeson` above

File: cardano-api/src/Cardano/Api/Value.hs [hunk 2]
```diff
@@ -84,6 +85,7 @@ import           Cardano.Api.SerialiseCBOR
 import           Cardano.Api.SerialiseRaw
 import           Cardano.Api.SerialiseUsing
 import           Cardano.Api.Utils (note)
+import qualified Data.Aeson.KeyMap as KeyMap
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Value.hs:88]: This import should be with `Data.Aeson` above

File: cardano-cli/src/Cardano/CLI/Run/Friendly.hs [hunk 3]
```diff
@@ -30,6 +30,7 @@ import           Cardano.Ledger.Crypto (Crypto)
 import qualified Cardano.Ledger.Shelley.API as Shelley
 
 import           Cardano.CLI.Helpers (textShow)
+import qualified Data.Aeson.Key as Aeson
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Run/Friendly.hs:33]: This import should be  with `Data.Aeson` above

File: cardano-cli/src/Cardano/CLI/Shelley/Output.hs [hunk 4]
```diff
@@ -26,6 +26,7 @@ import           Data.Word
 import           Cardano.CLI.Shelley.Orphans ()
 import           Cardano.Ledger.Shelley.Scripts ()
 import           Cardano.Slotting.Time (SystemStart (..))
+import qualified Data.Aeson.Key as Aeson
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Output.hs:29]: This import should be with `Data.Aeson` above

File: cardano-node/cardano-node.cabal [hunk 5]
```diff
@@ -186,7 +186,6 @@ library
                       , transformers
                       , transformers-except
                       , typed-protocols
-                      , unordered-containers
```
Comment (Jimbo4350) [inline cardano-node/cardano-node.cabal:189]: :tada:

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]

**Review by Jimbo4350 — APPROVED** [review]

**Review by deepfire — APPROVED** [review]

**Review by JaredCorduan — APPROVED** [review]
@nc6 thank you for this huge effort!

**Review by coot — APPROVED** [review]
LGTM

**Comment by nc6** [comment]
bors merge

**Pull request merged**